lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2310,82 +2339,277 @@ balanceTransactionSpec = do
-                                    (fromMaybe mempty extraChange)
-                                ]
-                        conjoin
-                            [ property $ extraFee >= feeRequirementIncrease
-                                & counterexample ("fee requirement increased by "
-                                    <> show feeRequirementIncrease
-                                    <> " but the extra fee was just "
-                                    <> show extraFee
-                                    )
-                            , fromMaybe mempty extraChange <> extraFee
-                                === surplus
-                            ]
+                        (TxFeeAndChange fee0 [])
+                        `shouldBe`
+                        Right (TxFeeAndChange surplus [])
+
+        it "feeDelta + changeDeltas == surplus .&&. \
+           \feeDelta covers increase to fee requirement" $
+            property $
+                withMaxSuccess 10000
+                prop_distributeSurplusDelta_coversIncreaseToFeeRequirement
+
+prop_distributeSurplusDelta_coversIncreaseToFeeRequirement
+    :: Coin -> Coin -> [Coin] -> Property
+prop_distributeSurplusDelta_coversIncreaseToFeeRequirement
+    surplus fee0 change0 =
+    counterexample (show mres) $ case mres of
+        Left _ ->
+            label "unable to distribute surplus" $
+                property (surplus < (maxCoinCost <> maxCoinCost))
+        Right (TxFeeAndChange feeDelta changeDeltas) -> do
+            let feeRequirementIncrease = mconcat
+                    [ costOfIncreasingCoin feePolicy fee0 feeDelta
+                    , F.fold $ zipWith (costOfIncreasingCoin feePolicy)
+                        change0
+                        changeDeltas
+                    ]
+            conjoin
+                [ property $ feeDelta >= feeRequirementIncrease
+                    & counterexample ("fee requirement increased by "
+                        <> show feeRequirementIncrease
+                        <> " but the fee delta was just "
+                        <> show feeDelta
+                        )
+                , F.fold changeDeltas <> feeDelta
+                    === surplus
+                ]
+  where
+    feePolicy = LinearFee LinearFunction { intercept = 0, slope = 44 }
+    mres = distributeSurplusDelta
+        feePolicy surplus (TxFeeAndChange fee0 change0)
+    maxCoinCost = maximumCostOfIncreasingCoin feePolicy
+
+--------------------------------------------------------------------------------
+-- Properties for 'distributeSurplus'
+--------------------------------------------------------------------------------
+
+instance Arbitrary FeePolicy where
+    arbitrary = do
+        intercept <- frequency
+            [ (1, pure 0.0)
+            , (3, getPositive <$> arbitrary)
+            ]
+        slope <- frequency
+            [ (1, pure 0.0)
+            , (3, getPositive <$> arbitrary)
```

Comment:
💯 thanks!

It might be nice to throw in a few mainnet values:
```suggestion
            , (3, getPositive <$> arbitrary)
            , (1, pure 44)
```

And also for the intercept:
```
            , (1, pure 150_000)
```

Reply:
Nice idea! Fixed in b7a7addda6ffd6f6c7b70450a5133c5115e42e83.